src/vmspawn/vmspawn.c

```diff
@@ -668,6 +702,24 @@ static int start_virtiofsd(sd_bus *bus, const char *scope, const char *directory
         if (!cmdline)
                 return log_oom();
 
+        if (arg_uid_shift != UID_INVALID) {
+                r = strv_extend(&cmdline, "--uid-map");
```

Comment:
So since these options will make virtiofsd shell out to `newuidmap/newgidmap` I'm wondering if we should instead allow passing a userns fd to virtiofsd via `sd_listen()`. That way mkosi can take care of passing in a user namespace and vmspawn would run virtiofsd in that user namespace.

It'll also be required if we want to implement `--ephemeral` since we'll need to join the userns to be able to copy the files from the root directory.

Reply:
can user namespaces be passed to units?